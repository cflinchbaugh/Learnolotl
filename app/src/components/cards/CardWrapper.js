import React, { Component } from 'react';
import { connect } from 'react-redux';
import { revealOption } from '../../actions/cardActions';
import Card from './Card';


// TODO: Content min-height, is this necessary?
    // It's not lookin' so good on large-format monitors
class CardWrapper extends Component {
    constructor(props) {
        super(props);

        this.handleRevealClick = this.handleRevealClick.bind(this);
    }

    render() {
        let activeValue = this._processActiveValue(),
            translationsArr = this._processRevealOptions(),
            cardData = {
                activeValue,
                revealItem: this.props.revealItem,
                fadeInVanishOutAnimationData: {
                    timeout: 300,
                    classNames: "fade-in-vanish-out" 
                },
                revealOptionsData: {
                    display: this.props.flashCardDisplay,
                    optionsArray: translationsArr,
                    handleClickOption: this.handleRevealClick
                }
            }

        return (
            <Card {...cardData}/>
        )
    }

    _processRevealOptions() {
        let revealOptions = this.props.data.revealOptionData.filter( (item) => {
            return item.id !== this.props.mode;
        });

        return revealOptions;
    }

    handleRevealClick(e) {
        let i = 0,
            translationsArr = this.props.data.revealOptionData,
            updatedRevealOptionData = [];

        for ( ; i < translationsArr.length; i++) {
            if (translationsArr[i].id === e) {
                updatedRevealOptionData.push(translationsArr[i]);
            }
        }

        this.props.revealOption(updatedRevealOptionData);
    }

    _processActiveValue() {
        let activeCardData;

        // Loop over all revealOptionData, 
            // return the one currently active 
            // (determined by current MODE)
        activeCardData = this.props.data.revealOptionData.find(function(obj) {
            return obj.id.toLowerCase() === this.props.mode.toLowerCase();
        }, this);

        //Return the value
        return activeCardData.value;
    }

   
}

const mapStateToProps = state => ({
    revealItem: state.cards.revealOption,
    flashCardDisplay: state.cards.flashCardsDisplay
});

export default connect(mapStateToProps, { revealOption })(CardWrapper);