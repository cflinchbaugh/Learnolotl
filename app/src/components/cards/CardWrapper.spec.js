import React from 'react';
import {shallow} from 'enzyme';
import {CardWrapper} from './CardWrapper';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('CardWrapper', () => {
    it('renders and matches snapshot', () => {
        const renderer = new ShallowRenderer();

        const cardWrapperData = {
            revealItem: [],
            data: {
                revealOptionData: [
                    {
                        id: 'English'
                    }
                ]
            },
            mode: 'English'
        }

        const tree = renderer.render(
            <CardWrapper {...cardWrapperData}/>
        );

        expect(tree).toMatchSnapshot();
    });

});