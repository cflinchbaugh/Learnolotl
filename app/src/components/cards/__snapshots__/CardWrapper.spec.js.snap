// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardWrapper renders and matches snapshot 1`] = `
<Card
  activeValue={undefined}
  fadeInVanishOutAnimationData={
    Object {
      "classNames": "fade-in-vanish-out",
      "timeout": 300,
    }
  }
  revealItem={Array []}
  revealOptionsData={
    Object {
      "display": undefined,
      "handleClickOption": [Function],
      "optionsArray": Array [],
    }
  }
/>
`;
